### PR TITLE
drivers: current sense amp: add pm

### PR DESCRIPTION
**Pull request by nixward**
Enable power management for current amplifier sensor.

Also adds the `adc_read_dt()` API

### CODE REVIEW

**Comment by nixward** [comment]
@nandojve could you please have a look

File: drivers/sensor/current_amp/current_amp.c [hunk 1]
```diff
@@ -8,27 +8,36 @@
...
 #include <zephyr/drivers/adc/current_sense_amplifier.h>
+#include <zephyr/drivers/gpio.h>
 #include <zephyr/drivers/sensor.h>
+#include <zephyr/pm/device.h>
 
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(current_amp, CONFIG_SENSOR_LOG_LEVEL);
 
+struct current_amp_config {
+	struct current_sense_amplifier_dt_spec current;
```
Comment (nandojve) [inline drivers/sensor/current_amp/current_amp.c:19]: Not a block but for me when I read the code it will be more clear if I read `sensor` or `current_sensor`.

File: include/zephyr/drivers/adc.h [hunk 2]
```diff
@@ -681,6 +681,21 @@ static inline int z_impl_adc_read(const struct device *dev,
...
+/**
+ * @brief Set a read request from a struct adc_dt_spec.
+ *
+ * @param spec ADC specification from Devicetree.
+ * @param sequence  Structure specifying requested sequence of samplings.
+ *
+ * @return A value from adc_read().
+ * @see adc_read()
+ */
+static inline int adc_read_dt(const struct adc_dt_spec *spec,
```
Comment (nandojve) [inline include/zephyr/drivers/adc.h:693]: This I think should be in another PR because start to mix the goals.

**Review by nandojve — CHANGES_REQUESTED** [review]

File: include/zephyr/drivers/adc.h (hunk 2 above)
Comment (nixward) [reply include/zephyr/drivers/adc.h:693]: Moved here https://github.com/zephyrproject-rtos/zephyr/pull/63010

File: drivers/sensor/current_amp/current_amp.c (hunk 1 above)
Comment (nixward) [reply drivers/sensor/current_amp/current_amp.c]: `current_sense` ok?

File: drivers/sensor/current_amp/current_amp.c [hunk 3]
```diff
@@ -8,27 +8,36 @@
...
 
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(current_amp, CONFIG_SENSOR_LOG_LEVEL);
 
+struct current_amp_config {
+	struct current_sense_amplifier_dt_spec current_sense;
+#ifdef CONFIG_PM_DEVICE
+	struct gpio_dt_spec gpio_power;
+#endif
+};
```
Comment (fabiobaltieri) [inline drivers/sensor/current_amp/current_amp.c:23]: how about having the gpio part of `struct current_sense_amplifier_dt_spec` instead? It is defined to match the binding so I think it would make more sense that way. One extra pointer all the time if you don't need it but then it's in flash so prob not a big deal.

Comment (nixward) [reply drivers/sensor/current_amp/current_amp.c]: Done 👍

File: drivers/sensor/current_amp/current_amp.c (hunk 1 above)
Comment (nixward) [reply drivers/sensor/current_amp/current_amp.c]: @nandojve with the [below](https://github.com/zephyrproject-rtos/zephyr/pull/62992#discussion_r1334961553) this field is now removed.

File: drivers/sensor/current_amp/current_amp.c [hunk 4]
```diff
@@ -73,6 +75,38 @@ static const struct sensor_driver_api current_api = {
...
+	if (config->power_gpio.port == NULL) {
+		LOG_ERR("PM not supported");
+		return -ENOTSUP;
+	}
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		ret = gpio_pin_set_dt(&config->power_gpio, 1);
+		if (ret != 0) {
+			LOG_ERR("failed to set GPIO for PM resume");
```
Comment (fabiobaltieri) [inline drivers/sensor/current_amp/current_amp.c:93]: How about being explicit and `return ret` here and `return 0` at the end? Maybe bit more on the personal preference side but explicit exits are nice and unambiguous.

Comment (nixward) [reply drivers/sensor/current_amp/current_amp.c:93]: Done

**Review by fabiobaltieri — APPROVED** [review]

**Review by nandojve — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
